pytrainer/test/lib/test_gpx.py

```diff
@@ -109,5 +121,141 @@ def test_missing_name(self):
+                self.assertEquals(0.35416037011805446, total_dist)
+                self.assertEquals(4.0                , total_time)
+                self.assertEquals(318.7466890419602  , maxvel)
+                self.assertEquals(83                 , maxhr)
+        except():
+            self.fail()
+
+    def test_getDate(self):
+        try:
+            for xml_file_path in self.xml_files:
+                gpx = Gpx(None, xml_file_path)
+                date = gpx.getDate()
+                self.assertEquals('2018-03-25', date)
+        except():
+            self.fail()
+
+    def test_getTrackRoutes(self):
+        try:
+            for xml_file_path in self.xml_files:
+                gpx = Gpx(None, xml_file_path)
+                tracks = gpx.getTrackRoutes()
+                self.assertEquals([('Evening Ride', '2018-03-25')], tracks)
+        except():
+            self.fail()
+
+    def test_getUnevenness(self):
+        try:
+            for xml_file_path in self.xml_files:
+                gpx = Gpx(None, xml_file_path)
+                upositive, unegative = gpx.getUnevenness()
+                self.assertEquals(2.6000000000000227, upositive)
+                self.assertEquals(0.0               , unegative)
+        except():
+            self.fail()
+
+    def test_getTrackList(self):
+        try:
+            for xml_file_path in self.xml_files:
+                gpx = Gpx(None, xml_file_path)
+                values = gpx.getTrackList()
+                if values[0][2] == 0.0:
+                    # workaround: in some cases the total_time is not provided
+                    track_mod = [e[:2]+(0.0,)+e[3:] for e in self.reference_track]
+                    self.assertEquals(track_mod, values)
+                else:
+                    self.assertEquals(self.reference_track, values)
+        except():
+            self.fail()
+
+    def test_getHeartRateAverage(self):
+        try:
+            for xml_file_path in self.xml_files:
+                gpx = Gpx(None, xml_file_path)
+                avg_hr = gpx.getHeartRateAverage()
+                self.assertEquals(79, avg_hr)
+        except():
+            self.fail()
+
+    def test_getCalories(self):
+        try:
+            for xml_file_path in self.xml_files:
+                gpx = Gpx(None, xml_file_path)
+                calories = gpx.getCalories()
+                self.assertEquals(0, calories)
+        except():
+            self.fail()
+
+    def test_getStart_time(self):
+        try:
+            for xml_file_path in self.xml_files:
+                gpx = Gpx(None, xml_file_path)
+                start_time = gpx.getStart_time()
+                self.assertEquals('19:13:17', start_time)
+        except():
+            self.fail()
+
+    def test_getLaps(self):
+        try:
+            for xml_file_path in self.xml_files:
+                gpx = Gpx(None, xml_file_path)
+                laps = gpx.getLaps()
```

Comment:
At least the return value of Gpx.getLaps should change from list to an iterator, so please cast to list() here.

The same may apply to others, but I haven't checked.